tests/units/oup/test_oup_repository.py

```diff
@@ -0,0 +1,73 @@
+from io import BytesIO
+from unittest.mock import MagicMock, patch
+
+import pytest
+from oup.repository import OUPRepository
+
+
+class S3BucketResultObj:
+    def __init__(self, key) -> None:
+        self.key = key
+
+
+S3_RETURNED_VALUES = [
+    "01_content.pdf",
+    "01_content.xml",
+    "02_content.pdf",
+    "02_content.xml",
+    "03_some_random_file.img",
+    "text_file.txt",
+]
+
+FIND_ALL_EXTRACTED_FILES_EXPECTED_VALUES = [
+    "01_content.pdf",
+    "01_content.xml",
+    "02_content.pdf",
+    "02_content.xml",
+]
+
+FIND_ALL_EXPECTED_VALUES = [
+    {"xml": "01_content.xml", "pdf": "01_content.pdf"},
+    {"xml": "02_content.xml", "pdf": "02_content.pdf"},
+]
+
+expected_file = BytesIO()
+
+
+@pytest.fixture
+def boto3_fixture():
+    with patch("common.s3_service.boto3", autospec=True) as boto3_mock:
+        boto3_mock.resource.return_value.Bucket.return_value.objects.filter.return_value.all.return_value = [
+            S3BucketResultObj(file) for file in S3_RETURNED_VALUES
+        ]
+        yield boto3_mock
+
+
+def test_find_all(boto3_fixture):
+    repo = OUPRepository()
+    assert repo.find_all() == FIND_ALL_EXPECTED_VALUES
+
+
+def test_find_all_extracted_files(boto3_fixture):
+    repo = OUPRepository()
+    assert (
+        repo._OUPRepository__find_all_extracted_files()
```

Comment:
I'm very confused, what is this?

Reply:
it is testing, does the [__find_all_extracted_files](https://github.com/cern-sis/workflows/blob/oup/dags/oup/repository.py#L52) method works as expected, it mean, does it returns just XML files

Reply:
but this is a private method, it should be test it the same way we are using it i.e. the `find_all()`

Reply:
But in find_all, we are getting an array of dicts with xml and pdf. And I want to check is the method gives me just xmls. The test for find_all is here:https://github.com/cern-sis/workflows/blob/fa015c884473cfb169998a172ca810e73e35145a/tests/units/oup/test_oup_repository.py#L46-L48

Or you had in mind something else?

Reply:
the test for find_all is enough?